### PR TITLE
TW-1605: Fix display thumbnail when user sent a video

### PR DESCRIPTION
**Pull request by nqhhdev**
### Ticket

- #1605 

### Need to merge:

- https://github.com/linagora/matrix-dart-sdk/pull/53

### Root cause:

- While sending a video, the thumbnail display condition was wrong. This condition says that only when uploading and encrypting will the thumbnail be displayed.

```dart

bool isThumbnailGenerated() {
    return fileSendingStatus == FileSendingStatus.encrypting ||
        fileSendingStatus == FileSendingStatus.uploading;
  }

```

### Solution:

- Delete the false condition, update the new condition 

```dart
matrixFile.bytes != null && matrixFile.bytes!.isNotEmpty
```

### Resolved

https://github.com/linagora/twake-on-matrix/assets/99852347/244df7ae-0102-48af-94a7-8ff8a56ed9ba

### CODE REVIEW

File: lib/presentation/mixins/send_files_mixin.dart [hunk 1]
```diff
@@ -41,6 +41,7 @@ mixin SendFilesMixin {
     final result = await FilePicker.platform.pickFiles(
       withReadStream: true,
       allowMultiple: true,
+      withData: true,
```
Comment (hoangdat) [inline lib/presentation/mixins/send_files_mixin.dart:44]: why we need it?

**Review by hoangdat — COMMENTED** [review]
what is the root cause?

File: lib/presentation/extensions/send_file_extension.dart [hunk 2]
```diff
@@ -455,35 +455,22 @@ extension SendFileExtension on Room {
...
-    }
-    var width = fileInfo.width;
-    var height = fileInfo.height;
-    if (width == null || height == null) {
-      final imageDimension = await runBenchmarked(
-        '_calculateImageDimension',
-        () => _calculateImageBytesDimension(fileInfo.imagePlaceholderBytes),
-      );
-      width = imageDimension.width.toInt();
-      height = imageDimension.height.toInt();
```
Comment (hoangdat) [inline lib/presentation/extensions/send_file_extension.dart:478]: try catch

**Review by Te-Z — APPROVED** [review]

File: lib/domain/model/extensions/platform_file/platform_file_extension.dart [hunk 3]
```diff
@@ -1,12 +1,13 @@
...
 import 'package:matrix/matrix.dart';
+import 'package:path_provider/path_provider.dart';
 
 extension PlatformFileListExtension on PlatformFile {
   MatrixFile toMatrixFile() {
     return MatrixFile.fromMimeType(
       bytes: bytes,
       name: name,
-      filePath: '',
+      filePath: path ?? '${getTemporaryDirectory()}/$name',
```
Comment (sherlockvn) [inline lib/domain/model/extensions/platform_file/platform_file_extension.dart:10]: await

**Review by sherlockvn — APPROVED** [review]

**Review by hoangdat — APPROVED** [review]

**Pull request merged**